feat(deps): Allow consumers to use newer uswds minor versions (#2600)
trussworks/react-uswds: README.md

### USWDS

It is strongly suggested applications use the same version of USWDS that was used to build the version of ReactUSWDS they're using (see this repo's uswds `devDependency` in [package.json](./package.json)).
A version mismatch may result in unexpected markup & CSS combinations.
For flexibility, ReactUSWDS will not trigger warnings if consumers choose to use a higher minor version of `uswds` (hence the careted uswds `peerDependency` in [package.json](./package.json)).

> If encountering unexpected markup issues when choosing _not_ to use the matching `devDependency` version of `uswds`, consumers should always verify that aligning the versions does not resolve their issue(s).

Reviewer: ```suggestion
If encountering unexpected markup issues when choosing _not_ to use the matching `devDependency` version of `uswds`, consumers should check whether aligning the versions resolves their issue(s).
```
